### DISCUSSION TITLE
Trigger RepositoryDispatch with custom payload?

### DISCUSSION THREAD

**Discussion by rtsao** [community/community#24832] [Apps, API and Webhooks]
<p>It seems there is currently no way <a href="https://developer.github.com/actions/managing-workflows/triggering-a-repositorydispatch-webhook/" rel="nofollow noopener">trigger a RepositoryDispatch webhook</a> with a custom payload (i.e. the only field is “event type”).</p>
<p>Of course, this could be encoded into the event type as a suffix or something (e.g. {“event_type”: “my_event_name.some_data_here”}) but this seems far from ideal.</p>
<p>It would be really nice to be able to attach arbitrary metadata to RepositoryDispatch events, which could be consumable by consumers of those events.</p>

**Comment by lee-dohm** [comment]
<p>Thanks for this feedback! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Please let me know if you have any other questions.</p>

**Comment by rtsao** [comment, marked as answer]
<p>It seems this has been added?</p>
<p><a href="https://developer.github.com/v3/repos/#input-3" rel="nofollow noopener">https://developer.github.com/v3/repos/#input-3</a></p>
